src/main/services/additional-content/maps/local-maps-manager.service.ts

```diff
@@ -306,6 +309,85 @@ export class LocalMapsManagerService {
         return null;
     }
 
+    public importMaps(zipPaths: string[], version?: BSVersion): Observable<Progression> {
+        const progress: Progression<string> = {
+            total: zipPaths.length,
+            current: 0,
+            data: "",
+        };
+
+        return new Observable<Progression<string>>(observer => {
+            (async () => {
+                for (const zipPath of zipPaths) {
+                    ++progress.current;
+                    progress.data = zipPath;
+                    observer.next(progress);
+
+                    try {
+                        await this.importMap(zipPath, version);
+                    } catch (error: any) {
+                        log.error(`Could not import "${zipPath}"`, error);
+                        this.ipc.send<{map: string, error: string}>(
```

Comment:
Its preferable to remove this `ipc.send` cause' it can spam the user if all dropped zips are invalid.\
Just handle errors on the frontend as normally and show a warning if the progression does not reach 100%
(This is even truer when this function will handle zips of multiple maps)